src/runtime/object.cpp

```diff
@@ -1593,6 +1609,19 @@ extern "C" LEAN_EXPORT object * lean_int_big_div(object * a1, object * a2) {
     }
 }
 
+extern "C" LEAN_EXPORT object * lean_int_big_div_exact(object * a1, object * a2) {
+    if (lean_is_scalar(a1)) {
+        int n = lean_scalar_to_int(a1);
+        return n == 0 ? a1 : lean_box(static_cast<unsigned>(-1));
```

Comment:
Future readers of this code might be happy about a comment here that explains why this works (this code path is only taken if `a1` is scalar but `a2` isn't, we globally guarantee that non-scalar values never contain anything that could fit in a scalar, and `a2` divides `a1`, which can only happen if either `a1` is zero or `a1` is `MIN_INT` and `a2` is `MAX_INT + 1 = -MIN_INT`).